src/lib.rs

```diff
@@ -65,6 +73,14 @@ impl Wiki {
         }
     }
 
+    #[private]
```

Comment:
Is this modifier required? A private modifier means it can only be called by the contract owner. I don't think as a contract owner, we are interested in calling this endpoint. It is simply a pure Rust function.

Reply:
Yeah, I was reading about the `#[private]` macro from [this page](https://github.com/near/near-sdk-rs#features). I assumed that by default, all methods are callable by another smart contract, so if we have methods that we know should only be called by the same smart contract, then we can use this macro to ensure it. Do you have the same understanding though? should we test by writing another smart contract and try calling this method from there?

Reply:
No we should not test private because it has already been tested by the NEAR team. I'm say that we don't plan to do any call to `vote`, only to upvote and downvote, so we can remove this modifier. Because this function is not `pub`, it is not accessible by anyone by from within the contract itself.
The rules are:
- `pub` function can be called by any one
- `pub` function with `private` modifier can be call only by owner
- non-`pub` function is not accessible from the outside at all, and hence we don't need the private modifier

Reply:
makes sense. Thanks for the elaboration :) I'll remove the `#[private]` macro where we don't need it then.

Reply:
And yes, it was also documented clearly here: https://www.near-sdk.io/contract-structure/near-bindgen#private-methods

So the use of the `#[private]` macro is that specific.